nupic/research/frameworks/pytorch/hooks/base.py

```diff
@@ -0,0 +1,93 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import abc
+
+
+class TrackStatsHookBase(metaclass=abc.ABCMeta):
+    """
+    This is a base class for tracking `hooks`_ via forward or backward passes. This
```

Comment:
is the underscore on this line a typo?

Reply:
It's the proper format (as recommended by [PEP](https://www.python.org/dev/peps/pep-0287/)) for a link according to the [RestructuredText](https://www.writethedocs.org/guide/writing/reStructuredText/) markup language. We have this in other parts of our [code](https://github.com/numenta/nupic.torch/blob/0f0e2452dd648ef0421d53d872c5aaa4b6a7feae/nupic/torch/models/sparse_cnn.py#L164) as well.